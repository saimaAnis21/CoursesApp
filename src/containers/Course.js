import React, { useState, useEffect } from 'react';
import CourseAddForm from '../components/CourseAddForm';
import CourseDisplay from '../components/CourseDisplay';
import { GetAuthToken, GetEmail } from '../logic/localStorage';
import '../styles.css';

const Course = () => {
  const [hasToken, setHasToken] = useState();
  const [email, setEmail] = useState();

  useEffect(() => {
    const token = GetAuthToken();
    if (token === null) {
      setHasToken(false);
    } else {
      setHasToken(true);
      setEmail(GetEmail());
    }
  });

  return (
    <div className="txt-align mt-10">

      { hasToken ? (
        <p className="text-success">
          {email}
          {' '}
          is logged in!!
        </p>
      ) : <p>Please Sign in or Sign up if you do not have an account!!</p>}
      { hasToken ? <CourseAddForm /> : <p><a href="/signin">Sign in!</a></p>}
      { hasToken ? <CourseDisplay /> : (
        <p><a href="/signup">Sign up!</a></p>
      )}
    </div>
  );
};

export default Course;
