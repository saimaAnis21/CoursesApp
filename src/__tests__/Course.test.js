import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Course from '../containers/Course';

const store = createStore(() => [], {});
const Wrapper = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);

Wrapper.propTypes = {
  children: PropTypes.shape({}).isRequired,
};

describe('Course', () => {
  it('renders correctly', () => {
    const tree = render(<Course />, { wrapper: Wrapper });
    expect(tree).toMatchSnapshot();
  });
});
