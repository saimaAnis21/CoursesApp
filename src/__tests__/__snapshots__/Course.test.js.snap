// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="txt-align mt-10"
      >
        <p>
          Please Sign in or Sign up if you do not have an account!!
        </p>
        <p>
          <a
            href="/signin"
          >
            Sign in!
          </a>
        </p>
        <p>
          <a
            href="/signup"
          >
            Sign up!
          </a>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="txt-align mt-10"
    >
      <p>
        Please Sign in or Sign up if you do not have an account!!
      </p>
      <p>
        <a
          href="/signin"
        >
          Sign in!
        </a>
      </p>
      <p>
        <a
          href="/signup"
        >
          Sign up!
        </a>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
